### PR TITLE
Add draft of PlatformIO post

### PR DESCRIPTION
**Pull request by franc0is**

### CODE REVIEW

File: _drafts/platformio.md [hunk 1]
```diff
@@ -0,0 +1,446 @@
+title: What About PlatformIO?
```
Comment (tyhoff) [inline _drafts/platformio.md:1]:
```suggestion
---
title: What About PlatformIO?
```

File: _drafts/platformio.md [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+description:
+    An overview of PlatformIO, as well as a getting started example on STM32.
+author: francois
+---
+
+A few weeks ago, I wrote about MCU SDKs and how frustrating I sometimes found it
```
Comment (tyhoff) [inline _drafts/platformio.md:7]:
```suggestion
A few weeks ago, [I wrote about MCU SDKs]({% post_url 2020-11-10-the-best-and-worst-mcu-sdks %}) and how frustrating I sometimes found it
```

File: _drafts/platformio.md [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+
+If you are able to piggy back on an RTOS, things look a bit better. For example,
+you can create a Zephyr project with PlatformIO for Atmel SAM, Freescale
+Kinetis, Nordic nRF5, NXP iMX RT, NXP LPC, Silabs EFM32, and STM32.
+
+You can use the PlatformIO client to list supported frameworks, platforms, and
+boards. The list changes regularly, as the project is very active. Here's the
+command you want to run:
+
+```
```
Comment (tyhoff) [inline _drafts/platformio.md:90]: Put new lines between each command.

File: _drafts/platformio.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+$ pio platform search
+# List frameworks
+$ pio platform frameworks
+# List boards
+$ pio boards
+```
+
+### Package Management
+
+PlatformIO hosts a registry of embedded library which can be added to every
```
Comment (tyhoff) [inline _drafts/platformio.md:101]:
```suggestion
PlatformIO hosts a registry of embedded libraries which can be added to every
```

File: _drafts/platformio.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+build_flags = -Wall -Wl,--gc-sections
+```
+
+Should you need to customize your build beyond what the `.ini` file provides,
+you will need to write SCons scripts which are invoked by the PlatformIO build
+environment. SCons scripts are written in Python.
+
+Two hooks are available for those scripts: a PRE hook which runs before the
+build executes, a POST hook which runs after the build. You can use these hooks
+to chaange the build instructions for individual files, pre-process some source
```
Comment (tyhoff) [inline _drafts/platformio.md:162]:
```suggestion
to change the build instructions for individual files, pre-process some source
```

File: _drafts/platformio.md [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+code, or post-process the build artifacts[^advanced_scripting].
+
+While some may chafe at the need to learn yet another build system - are for one
```
Comment (tyhoff) [inline _drafts/platformio.md:165]:
```suggestion
While some may chafe at the need to learn yet another build system - I for one
```

File: _drafts/platformio.md [hunk 7]
```diff
@@ -0,0 +1,468 @@
...
+still like my Makefiles - SCons is a mature and powerful build system.
+
+## Getting Started with the PlatformIO CLI
+
+My favorite way to use PlatformIO is the CLI, also known as "PlatformIO Core".
```
Comment (tyhoff) [inline _drafts/platformio.md:170]:
```suggestion
My favorite way to use PlatformIO is with the CLI, also known as "PlatformIO Core".
```

File: _drafts/platformio.md [hunk 8]
```diff
@@ -0,0 +1,468 @@
...
+Project has been successfully initialized! Useful commands:
+`pio run` - process/build project from the current directory
+`pio run --target upload` or `pio run -t upload` - upload firmware to a target
+`pio run --target clean` - clean project (remove compiled files)
+`pio run --help` - additional information
+```
+
+### Toolchains and linker scripts
+
+On many of my project, I use a specific version of the GCC toolchain, and to
```
Comment (tyhoff) [inline _drafts/platformio.md:220]:
```suggestion
On many of my projects, I use a specific version of the GCC toolchain, and to
```

File: _drafts/platformio.md [hunk 9]
```diff
@@ -0,0 +1,468 @@
...
+4. Link everything into an ELF
+5. Copy it into a BIN
+6. Print out RAM and Flash usage
+
+All with minimal configuration!
+
+### Unit Tests
+
+Regular readers of Interrupt will know that we are [big fans of unit
+testing XXX TODO](). PlatformIO has built in supports for running tests on your
```
Comment (tyhoff) [inline _drafts/platformio.md:341]:
```suggestion
Regular readers of Interrupt will know that we are [big fans of unit
testing]({% post_url 2019-10-08-unit-testing-basics %}). PlatformIO has built in supports for running tests on your
```

**Review by tyhoff — COMMENTED** [review]
First read through. Give me a sec to compile other feedback.

File: _drafts/platformio.md (hunk 3 above)
Comment (franc0is) [reply _drafts/platformio.md:91]: 👍

**Pull request merged**